aws/src/integration/java/org/apache/iceberg/aws/glue/GlueCatalogTableTest.java

```diff
@@ -0,0 +1,241 @@
+    List<TableIdentifier> tables = glueCatalog.listTables(Namespace.of(namespace));
+    Assert.assertEquals(1, tables.size());
+    Assert.assertEquals(TableIdentifier.of(namespace, tableName), tables.get(0));
+  }
+
+  @Test
+  public void testTableExists() {
+    String namespace = createNamespace();
+    String tableName = createTable(namespace);
+    Assert.assertTrue(glueCatalog.tableExists(TableIdentifier.of(namespace, tableName)));
+  }
+
+  @Test
+  public void testUpdateTable() {
+    String namespace = createNamespace();
+    String tableName = getRandomName();
+    // current should be null
+    TableOperations ops = glueCatalog.newTableOps(TableIdentifier.of(namespace, tableName));
+    TableMetadata current = ops.current();
+    Assert.assertNull(current);
+    // create table, refresh should update
+    createTable(namespace, tableName);
+    current = ops.refresh();
+    Assert.assertEquals(schema.toString(), current.schema().toString());
+    Assert.assertEquals(partitionSpec, current.spec());
+    Table table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    Assert.assertTrue(table.history().isEmpty());
+    // commit new version, should create a new snapshot
+    table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    DataFile dataFile = DataFiles.builder(partitionSpec)
+        .withPath("/path/to/data-a.parquet")
+        .withFileSizeInBytes(10)
+        .withRecordCount(1)
+        .build();
+    table.newAppend().appendFile(dataFile).commit();
+    table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    Assert.assertEquals(1, table.history().size());
+  }
+
+  @Test
+  public void testRenameTable() {
+    String namespace = createNamespace();
+    String tableName = createTable(namespace);
+    Table table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    // rename table
+    String newTableName = tableName + "_2";
+    glueCatalog.renameTable(TableIdentifier.of(namespace, tableName), TableIdentifier.of(namespace, newTableName));
+    Table renamedTable = glueCatalog.loadTable(TableIdentifier.of(namespace, newTableName));
+    Assert.assertEquals(table.location(), renamedTable.location());
+    Assert.assertEquals(table.schema().toString(), renamedTable.schema().toString());
+    Assert.assertEquals(table.spec(), renamedTable.spec());
+    Assert.assertEquals(table.currentSnapshot(), renamedTable.currentSnapshot());
+  }
+
+  @Test
+  public void testRenameTableRollback() {
+    String namespace = createNamespace();
+    String tableName = createTable(namespace);
+    TableIdentifier id = TableIdentifier.of(namespace, tableName);
+    Table table = glueCatalog.loadTable(id);
+    // rename rollback
+    AssertHelpers.assertThrows("should not have table",
+        NoSuchTableException.class,
+        () -> glueCatalog.renameTable(
+            TableIdentifier.of(namespace, tableName), TableIdentifier.of(namespace, tableName + "-3")));
```

Comment:
What is a rename rollback? Why does this test fail? The table looks like it should exist.

Reply:
Sorry the message is vague, and the test here is not very intuitive, I will update.

Because there is no actual `rename` operation in Glue, rename means deleting an old table and creating a new table in Glue pointing to the same metadata. If the creation process fails, it should "rollback" and keep the old table.